JSONObject.java

```diff
@@ -1465,51 +1461,84 @@ public static Writer quote(String string, Writer w) throws IOException {
         String hhhh;
         int i;
         int len = string.length();
+        int prev = 0;
 
-        w.write('"');
+        w.append('"');
         for (i = 0; i < len; i += 1) {
             b = c;
             c = string.charAt(i);
             switch (c) {
             case '\\':
             case '"':
-                w.write('\\');
-                w.write(c);
+                if(prev < i) {
```

Comment:
Appending sequence ranges here:

* If the target Appendable is unbuffered, a sequence will write more quickly than single characters.
* If the target Appendable is a StringBuffer or StringBuilder, writing sequences will help reduce buffer resizes

In any circumstance, writing a single character at a time, with a method call per character, is likely to be slower than gathering sequences where possible.